addons/binding/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/internal/handler/HomeAssistantThingHandler.java

```diff
@@ -89,10 +91,11 @@
     public HomeAssistantThingHandler(Thing thing, MqttChannelTypeProvider channelTypeProvider, int subscribeTimeout,
             int attributeReceiveTimeout) {
         super(thing, subscribeTimeout);
+        this.gson = new GsonBuilder().registerTypeAdapterFactory(HAConfigTypeAdapterFactory.INSTANCE).create();
```

Comment:
Can we not share the gson instance with the FACTORY one?

Reply:
Definitely! I missed this one (second function in CFactory)